lib/cisco_node_utils/bgp.rb

```diff
@@ -794,6 +795,7 @@ def route_distinguisher=(rd)
         Feature.nv_overlay_enable
         Feature.nv_overlay_evpn_enable
       end
+      return nil if @vrf.nil? || @vrf == 'default'
```

Comment:
This should be a check *before* the platform check (no point in enabling features if we're not going to actually use them).

Also, the convention is that this, being a setter method, should raise a Cisco::UnsupportedError, rather than being a no-op.